hack/preload-images/preload_images.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	profile      = "generate-preloaded-images-tar"
+	minikubePath = "out/minikube"
+)
+
+var (
+	kubernetesVersion = ""
+	tarballFilename   = ""
+)
+
+func init() {
+	if kv := os.Getenv("KUBERNETES_VERSION"); kv != "" {
```

Comment:
I suggest using os.Args or a flag over an environment variable if possible, as env variables are not very discoverable.